compiler/luci-interpreter/src/BuddyMemoryManager.cpp

```diff
@@ -0,0 +1,91 @@
+/*
+ * Copyright (c) 2021 Samsung Electronics Co., Ltd. All Rights Reserved
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "luci_interpreter/BuddyMemoryManager.h"
+
+namespace luci_interpreter
+{
+
+BuddyMemoryManager::BuddyMemoryManager(uint8_t *memory_start, int32_t memSize)
+{
+  int32_t p = powOf2(memSize);
+  memSize = 1 << p;
+
+  _start_block = (Block *)memory_start;
```

Comment:
better to use c++ style casts, they have well defined use cases, so they are easier to read and understand (except cases with multiple casts in a row =_=).
```suggestion
  _start_block = reinterpret_cast<Block *>(memory_start);
```

Reply:
done